src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -28,33 +28,105 @@
 
 namespace PrestaShopBundle\ApiPlatform\Provider;
 
+use ApiPlatform\Metadata\CollectionOperationInterface;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShopBundle\ApiPlatform\Converters\ConverterInterface;
 use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
+use Symfony\Component\Serializer\Exception\ExceptionInterface;
 use Symfony\Component\Serializer\Serializer;
 
 class QueryProvider implements ProviderInterface
 {
     public function __construct(
         private readonly CommandBusInterface $queryBus,
+        private readonly iterable $converters,
         private readonly Serializer $apiPlatformSerializer
     ) {
     }
 
+    /**
+     * @throws \ReflectionException
+     * @throws \Exception
+     * @throws ExceptionInterface
+     */
     public function provide(Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $extraProperties = $operation->getExtraProperties();
+        $queryClass = $operation->getExtraProperties()['query'] ?? null;
+        $filters = $context['filters'] ?? [];
+        $uriVariables = array_merge($uriVariables, $filters);
 
-        $query = $extraProperties['query'] ?? null;
-
-        if (null === $query) {
+        if (null === $queryClass) {
             throw new NoExtraPropertiesFoundException();
         }
 
-        $queryResult = $this->queryBus->handle(new $query(...$uriVariables));
+        $query = $this->apiPlatformSerializer->denormalize($uriVariables, $queryClass);
+
+        //Try to call setter on additional query params
```

Comment:
Does it mean the serializer is not able to call the setters? I thought it would handle it internally already since it's based on the PropertyAccessor component

Is it a leftover of the initial implementation that maybe is not needed anymore? If not I wonder if it wouldn't be interesting to search for a way to enhance the serializer (by proving it some extra or custom normalizer for example) instead of the this "manually" here

Reply:
We have no solution to make it automatically for the moment, we have to go forward ;)